### PR TITLE
Event "data_definitions.import.total" never gets dispatched

### PR DESCRIPTION
**Pull request by paulverdu**
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | yes
| Branch?          | master

In Importer.php the event "data_definitions.import.total" never gets get dispatched when $data is an object of type \Wvision\Bundle\DataDefinitionsBundle\Provider\TraversableImportDataSet.

Fixed this by adding Countable interface and setting the count on object creation.

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
works for me, what does @dkarlovi think?

File: src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php [hunk 1]
```diff
@@ -9,23 +9,27 @@
  * files that are distributed with this source code.
  *
  * @copyright  Copyright (c) 2016-2019 w-vision AG (https://www.w-vision.ch)
- * @license    https://github.com/w-vision/DataDefinitions/blob/master/gpl-3.0.txt GNU General Public License version 3 (GPLv3)
+ * @license    https://github.com/w-vision/DataDefinitions/blob/master/gpl-3.0.txt GNU General Public License version 3
```
Comment (dkarlovi) [inline src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php:12]: Unrelated change?

File: src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php [hunk 2]
```diff
@@ -9,23 +9,27 @@
...
-use Iterator;
+use Countable;
+use IteratorIterator;
+use Traversable;
 
-class TraversableImportDataSet implements ImportDataSetInterface
+class TraversableImportDataSet implements ImportDataSetInterface, Countable
 {
-    private \IteratorIterator $iterator;
+    private IteratorIterator $iterator;
```
Comment (dkarlovi) [inline src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php:26]: Why CS changes in this PR?

File: src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php [hunk 3]
```diff
@@ -9,23 +9,27 @@
...
+    private int $count;
 
-    public function __construct(\Traversable $iterator)
+    public function __construct(Traversable $iterator)
     {
-        $this->iterator = new \IteratorIterator($iterator);
+        $this->count = iterator_count($iterator);
```
Comment (dkarlovi) [inline src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php:31]: Can we make this lazy? Only count it if the count is requested.

**Review by dkarlovi — CHANGES_REQUESTED** [review]
Small tweaks, otherwise good with me.

**Comment by dkarlovi** [comment]
@dpfaffenbauer looked at it, posted a review.

File: src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php (hunk 2 above)
Comment (paulverdu) [reply src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php:26]: My editor changes this. I'll change it back

File: src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php (hunk 1 above)
Comment (paulverdu) [reply src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php:12]: My editor changes this. I'll change it back

File: src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php (hunk 3 above)
Comment (paulverdu) [reply src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php:31]: It's not possible to do a count on a iterator since you can only rewind ones. That's why im doing this in the constructor before the IteratorIterator object is created. The count is also always requested in the importer.

**Review by dkarlovi — APPROVED** [review]

**Comment by paulverdu** [comment]
@dpfaffenbauer If we could merge this one as well. Would be awesome 😄
